Error getting value when using lambda expressions

When the serialized object is created using lambda expression, a `JsonbException` with a message of the sort "Error getting value on: com.acme.Entry$$Lambda$40/0x00000008000f7040@45018215" is thrown.

The following code:
```java
package com.acme;

import javax.json.bind.Jsonb;
import javax.json.bind.JsonbBuilder;

public class Entry {

    public interface Cat {
        String getName();
    }
    
    public static void main(String args[]) {
        Jsonb jsonb = JsonbBuilder.create();
        Cat cat = () -> "Catie";
        jsonb.toJson(cat);
    }
}
```

Results in the following stack trace:
```
SEVERE: Generating incomplete JSON
Exception in thread "main" javax.json.bind.JsonbException: Error getting value on: com.acme.Entry$$Lambda$40/0x00000008000f7040@2928854b
	at org.eclipse.yasson.internal.model.GetValueCommand.getValue(GetValueCommand.java:36)
	at org.eclipse.yasson.internal.model.ReflectionPropagation.getValue(ReflectionPropagation.java:72)
	at org.eclipse.yasson.internal.model.PropertyModel.getValue(PropertyModel.java:257)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.marshallProperty(ObjectSerializer.java:76)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:59)
	at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serialize(AbstractContainerSerializer.java:60)
	at org.eclipse.yasson.internal.Marshaller.serializeRoot(Marshaller.java:118)
	at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:76)
	at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:98)
	at com.acme.Entry.main(Entry.java:15)
Caused by: java.lang.IllegalAccessException: class org.eclipse.yasson.internal.model.GetFromGetter cannot access a member of class com.acme.Entry$$Lambda$40/0x00000008000f7040 with modifiers "public"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
	at java.base/java.lang.reflect.Method.invoke(Method.java:558)
	at org.eclipse.yasson.internal.model.GetFromGetter.internalGetValue(GetFromGetter.java:28)
	at org.eclipse.yasson.internal.model.GetValueCommand.getValue(GetValueCommand.java:34)
	... 9 more
```

Replacing the lambda expression with an anonymous class solves the issue. However, both expressions are supposed to be equivalent, and many IDEs suggest the shortening of anonymous classes into lambdas systematically.

--- a/src/main/java/org/eclipse/yasson/internal/model/PropertyValuePropagation.java
+++ b/src/main/java/org/eclipse/yasson/internal/model/PropertyValuePropagation.java
@@ -137,8 +137,8 @@ public abstract class PropertyValuePropagation {
         }
 
         Boolean accessible = isVisible(strategy -> strategy.isVisible(method), method.getDeclaringClass(), field, method, ctx);
-        //overridden by strategy, or anonymous class
-        if (accessible && (!Modifier.isPublic(method.getModifiers()) || method.getDeclaringClass().isAnonymousClass())) {
+        //overridden by strategy, anonymous class, or lambda
+        if (accessible && (!Modifier.isPublic(method.getModifiers()) || method.getDeclaringClass().isAnonymousClass() || method.getDeclaringClass().isSynthetic())) {
             overrideAccessible(method);
         }
         return accessible;

--- a/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Addressable.java
+++ b/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Addressable.java
@@ -1,0 +1,14 @@
+package org.eclipse.yasson.defaultmapping.lambda;
+
+/**
+ * A functional interface with no default properties.
+ */
+public interface Addressable {
+
+    /**
+     * Public readable property.
+     *
+     * @return name to be addressed with
+     */
+    String getName();
+}

--- a/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Cat.java
+++ b/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Cat.java
@@ -1,0 +1,18 @@
+package org.eclipse.yasson.defaultmapping.lambda;
+
+/**
+ * Class used to control serialization of lambda expression generated from functional interfaces with defaults.
+ */
+public class Cat implements Pet {
+
+    private String name;
+
+    Cat(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+}

--- a/src/test/java/org/eclipse/yasson/defaultmapping/lambda/LambdaExpressionTest.java
+++ b/src/test/java/org/eclipse/yasson/defaultmapping/lambda/LambdaExpressionTest.java
@@ -1,0 +1,40 @@
+package org.eclipse.yasson.defaultmapping.lambda;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Assert;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+/**
+ * Test marshalling objects generated using lambda expressions.
+ *
+ * Scenarios test lambda-generated objects serialization against concrete class objects serialization. It is thus
+ * assumed that concrete class behaviour is tested successfully elsewhere.
+ */
+public class LambdaExpressionTest {
+
+    private Jsonb jsonb;
+
+    @Before
+    public void before() {
+        jsonb = JsonbBuilder.create();
+    }
+
+    @Test
+    public void testMarshallFunctionalInterface() {
+        String name = "WALL-E";
+        Addressable control = new Robot(name);
+        Addressable lambda = () -> name;
+        Assert.assertEquals(jsonb.toJson(control), jsonb.toJson(lambda));
+    }
+
+    @Test
+    public void testMarshallFunctionalInterfaceWithDefaultProperties() {
+        String name = "Cheshire";
+        Pet control = new Cat(name);
+        Pet lambda = () -> name;
+        Assert.assertEquals(jsonb.toJson(control), jsonb.toJson(lambda));
+    }
+}

--- a/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Pet.java
+++ b/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Pet.java
@@ -1,0 +1,16 @@
+package org.eclipse.yasson.defaultmapping.lambda;
+
+/**
+ * A functional interface with one default property.
+ */
+public interface Pet extends Addressable {
+
+    /**
+     * Public readable property with default value.
+     *
+     * @return pet's age in months
+     */
+    default int getAge() {
+        return 0;
+    }
+}

--- a/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Robot.java
+++ b/src/test/java/org/eclipse/yasson/defaultmapping/lambda/Robot.java
@@ -1,0 +1,18 @@
+package org.eclipse.yasson.defaultmapping.lambda;
+
+/**
+ * Class used to control serialization of lambda expression generated from functional interfaces with no defaults.
+ */
+public class Robot implements Addressable {
+
+    private String name;
+
+    Robot(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+}

